src/compiler.rs

```diff
@@ -187,21 +187,38 @@ impl Compiler {
             .expect("Cannot find parent directory of file");
         self.include_paths.push(parent_dir.to_path_buf());
 
-        // should always be 0
-        let file_scope_id = ScopeId(project.current_module().id, 0);
-
         let file_id = match self.find_or_load_file(fname) {
             Ok(file_id) => file_id,
-            Err(e) => return (file_scope_id, Some(e)),
+            Err(e) => return (ScopeId(ModuleId(project.current_module_index), 0), Some(e)),
```

Comment:
Does the `file_id` here point to the prelude instead of the user's file (since you create the module for the user's file after)?

Reply:
It will point to the prelude. Technically this should point to the module that is importing the new module. Since the Rust compiler is deprecated, the refactor isn't necessary.

Reply:
We'll still need to get the Rust compiler to where it can handle multi-file support since without it would mean putting everything in one file for the selfhost compiler.

Reply:
I have a hack to add qualified names. I can add it to this PR. Which should fix using struct, classes, and enums from imported modules. That should be the last think that I know of that keeps multi-file support from working.

Reply:
That'd be great if you have that. I think once we have it landed and can start moving to multi-file projects, we can do that with selfhost and really move everyone over to contributing to selfhost instead of the rust-based compiler.